spec/controllers/api/encryptables_controller_spec.rb

```diff
@@ -99,6 +100,97 @@
       expect_json_object_includes_keys(credentials2_json_attributes, attributes)
       expect_json_object_includes_keys(credentials2_json_relationships, nested_models)
     end
+
+    context 'recent Credentials' do
+
+      let!(:recentCredentials) do
+        Fabricate.times(
+          6,
+          :credential,
+          folder: teams(:team1).folders.first,
+          team_password: Crypto::Symmetric::Aes256.random_key
+        )
+      end
+
+      it 'returns most recent cerdentials' do
+
+        login_as(:alice)
+
+        recentCredentials.each do |credential|
+          PaperTrail.request(whodunnit: alice.id) do
+            credential.touch
+          end
+        end
+
+        get :index, params: { recent: true }, xhr: true
+
+        expect(response.status).to be(200)
+        expect(data.size).to eq(5)
+        attributes = data.first['attributes']
+        expect(attributes['name']).to eq recentCredentials.last.name
+        expect(attributes['description']).to eq recentCredentials.last.description
+      end
+
+      it 'shows most recently used credential first in list' do
+        login_as(:alice)
+
+        PaperTrail.request(whodunnit: alice.id) do
+          recentCredentials.each do |credential|
+            credential.touch
```

Comment:
make sure you use the same function to create version entries as by controller read actions. 
is `touch` the right thing to do this?

Reply:
changed it to use the save_with_version method that is also used in the show action of the controller